### PR TITLE
Smoke test authority host

### PR DESCRIPTION
**Pull request by danieljurek**
This contains changes needed to get smoke tests working in other Azure clouds. 

This uses static resources today. Dynamic resource allocation is working locally but is failing for some reason in DevOps. I need to investigate further.

### CODE REVIEW

File: common/SmokeTests/SmokeTest/KeyVaultTest.cs [hunk 1]
```diff
@@ -32,7 +34,20 @@ public static async Task RunTests()
             string clientID = Environment.GetEnvironmentVariable("APP_CLIENT_ID");
             string clientSecret = Environment.GetEnvironmentVariable("CLIENT_SECRET");
             string keyVaultUri = Environment.GetEnvironmentVariable("KEY_VAULT_URI");
-            client = new SecretClient(new Uri(keyVaultUri), new ClientSecretCredential(tenantID, clientID, clientSecret));
+            
+            var authorityHost = Environment.GetEnvironmentVariable("AZURE_AUTHORITY_HOST")
```
Comment (weshaggard) [inline common/SmokeTests/SmokeTest/KeyVaultTest.cs]: Should we just require this even for the default cases now?

Comment (jongio) [reply common/SmokeTests/SmokeTest/KeyVaultTest.cs]: What do you mean by that wes?

**Review by jongio — APPROVED** [review]

File: common/SmokeTests/SmokeTest/KeyVaultTest.cs (hunk 1 above)
Comment (danieljurek) [reply common/SmokeTests/SmokeTest/KeyVaultTest.cs]: In the interest of getting this merged I don't think this is a blocker. We currently have a "safe" default to the public cloud for 3 languages for the smoke test including this one. I've opened this issue so we can close up that discussion: https://github.com/Azure/azure-sdk/issues/1104

**Pull request merged**